README.md

```diff
@@ -1,100 +1,105 @@
-
-```bash
-docker commit <NAME_OF_CONTAINER> <NAME_OF_NEW_IMAGE> && docker run -dti -v <NAME_OF_DOCKER_VOLUME>:<WORKDIR_OF_THE_CONTAINER> --name <NAME_OF_NEW_CONTAINER> <NAME_OF_NEW_IMAGE>"
-```
+# development-env
+
+A dev container based on Neovim.
+
+The expectation is that an container app under development will shared it's code through a volume.
+
+The dev container can then also be mounted to that volume. This then means the dev container can edit the application code, and make use of any dev dependencies which are available in the node_modules folder for example.
+
+The use case is currently limited to nodejs projects, hence the reference to node_modules above, but I will not prematurely optimise for other usecases at this point.
+
+# Usage
+
+## For container app development
+
+The primary usage comes from running a container from the image, with the appropriate docker volume specified, then `exec`ing in to the container and running neovim.
+
+Prior to this step you will need to have mounted the application code from the actual container app under development on to a docker volume.
+
+```bash
+docker run -d -v <NAME-OF-DOCKER-VOLUME>:/app --name <NAME-OF-CONTAINER> jslog/development-env:v<SEMVER-ID>
+docker exec -it <NAME_OF_CONTAINER> /bin/bash
+# During the first entry to the container you will want to run the Post-container-creation actions described below
+```
+It is suggested that you maintain a dedicated container for the project you are working on, and the volume which it's attached to.
+
+The benefit of this is that you don't have to worry about switching between volumes which can be a bit cumbersome, and you only have to do the Post-container-creation actions once per container.
+
+If the container has been stopped for any reason, you can simply start it again and as long as the original volume is still in place you can carry on where you left off.
+
+```bash
+docker start <NAME_OF_CONTAINER>
+docker exec -it <NAME_OF_CONTAINER> /bin/bash
+```
+
+
+## For development of the development-env project itself
+
+Updating the development environment project itself doesn't require mounting docker volumes since the application code and dev tools are already in the same place.
+
+```bash
+docker run -d --name <NAME-OF-CONTAINER> jslog/development-env:v<SEMVER-ID>
+docker exec -it <NAME_OF_CONTAINER> /bin/bash
+# During the first entry to the container you will want to run the Post-container-creation actions described below
+cd /development-env
+# Now make whatever modifications you want to the project, and push.
+# To test the changes you will need to:
+#  - pull the changes on to the project on the host system
+#  - run the usual build process described below
+#  - create a new container with a new name and test what you find inside
+```
+
+## Build process
+
+### Clone
+
+```bash
+git clone --config core.autocrlf=input --recursive git@github.com:js-jslog/development-env.git
```

Comment:
Change this to `git clone --config core.autocrlf=input --recursive https://github.com/js-jslog/development-env.git`. That's what's been tested.